### ISSUE TITLE
Rule blocked: CORERULES-203 CORERULES-204

### ISSUE THREAD

**Issue by apalmercdisc** [cdisc-org/cdisc-rules-engine#279]
**Links to related JIRA Tickets**
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-203
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-204

**Rule Information**
- SDTMIG:
- CG0332, CG0333:
- The first two characters of the dataset name must equal a domain that is present in the study
- The characters of the dataset name after 'FA' equal a domain that present in the study

**Describe the problem**
For CG0332 - look in the DOMAIN variable, the value should be the first 2 letters of the dataset name for the split datasets
For CG0333 - the 3rd and 4th characters of the dataset name should match the values from the DOMAIN variable in other datasets for the study.

**Describe the solution**
There needs to be an operation to list of domain codes coming from the value of the domain variable in all of the datasets selected for validation. study_domains (new operator)
This also requires new operators - prefix_is_not_contained_by, prefix_is_contained_by, suffix_is_not_contained_by, suffix_is_contained_by

**Proposed rule logic**
```yaml
Rule Type: Dataset metadata check
Check:
  all:
    - name: dataset_name
      operator: matches_regex
      value: ^[A-Za-z0-9]{3,4}
    - name: dataset_name
      operator: not_prefix_matches_regex
      prefix: 2
      value: (AP|FA)
    - name: dataset_name
      operator: prefix_is_not_contained_by (new operator)
      prefix: 2
      value: $domains_in_study
operations:
 - id: $domains_in_study
    operator: study_domains (new operator)
```

**Screenshots**
If applicable, add screenshots to help explain your problem and solution.
